optuna/storages/rdb/storage.py

```diff
@@ -326,14 +326,53 @@ def get_all_study_summaries(self):
 
         return study_sumarries
 
-    def create_new_trial(self, study_id):
-        # type: (int) -> int
+    def create_new_trial(self, study_id, template_trial=None):
+        # type: (int, Optional[structs.FronzenTrial]) -> int
 
         session = self.scoped_session()
 
-        trial = models.TrialModel(study_id=study_id, state=structs.TrialState.RUNNING)
+        if template_trial is None:
+            trial = models.TrialModel(study_id=study_id, state=structs.TrialState.RUNNING)
+        else:
+            # Because only `RUNNING` trials can be updated,
+            # we temporarily set the state of the new trial to `RUNNING`.
+            # After all fields of the trial have been updated,
+            # the state is set to `template_trial.state`.
+            temp_state = structs.TrialState.RUNNING
+
+            trial = models.TrialModel(
+                study_id=study_id,
+                state=temp_state,
+                value=template_trial.value,
+                datetime_start=template_trial.datetime_start,
+                datetime_complete=template_trial.datetime_complete,
+            )
 
         session.add(trial)
+        session.flush()
```

Comment:
Please add a comment to describe the reason of this `flush`.

Reply:
Added: https://github.com/pfnet/optuna/pull/464/commits/db3df2e75aa3187b82a607f75e4162647f0a1932